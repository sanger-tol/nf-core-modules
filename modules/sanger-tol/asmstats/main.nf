process ASMSTATS {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/c4/c47c4b4f55ce62f79edd9b4c5a26b7b1ed60684452dbbbd03f0be0d56c7b00be/data':
        'community.wave.seqera.io/library/seqtk_perl:37201934bb74266e' }"

    input:
    tuple val(meta), path(assembly)

    output:
    tuple val(meta), path("*.stats"), emit: stats
    path "versions.yml"             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    // WARNING: This module includes the asmstats script as a module binary in
    // ${moduleDir}/resources/usr/bin/asmstats. To use this module, you will
    // either have to copy this file to ${projectDir}/bin or set the option
    // nextflow.enable.moduleBinaries = true in your nextflow.config file.
    def args    = task.ext.args ?: ''
    def prefix  = task.ext.prefix ?: "${assembly.getName()}"
    def VERSION = "1.0.0"
    """
    asmstats \\
        $args \\
        ${assembly} \\
        > ${prefix}.stats

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        asmstats: ${VERSION}
        seqtk: \$(seqtk |& sed '/Version/!d; s/.* //')
    END_VERSIONS
    """

    stub:
    def args    = task.ext.args ?: ''
    def prefix  = task.ext.prefix ?: "${meta.id}"
    def VERSION = "1.0.0"
    """
    touch ${prefix}.stats

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        asmstats: \$(asmstats --version)
        seqtk: \$(seqtk |& sed '/Version/!d; s/.* //')
    END_VERSIONS
    """
}
