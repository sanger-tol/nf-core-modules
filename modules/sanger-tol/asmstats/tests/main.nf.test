nextflow_process {

    name "Test Process ASMSTATS"
    script "../main.nf"
    process "ASMSTATS"

    tag "modules"
    tag "modules_sangertol"
    tag "asmstats"

    config "./nextflow.config"

    test("meles meles - fasta") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'Meles_meles/assembly/release/mMelMel3.1_paternal_haplotype/GCA_922984935.2.subset.fasta.gz', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("meles meles - fasta - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'Meles_meles/assembly/release/mMelMel3.1_paternal_haplotype/GCA_922984935.2.subset.fasta.gz', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
