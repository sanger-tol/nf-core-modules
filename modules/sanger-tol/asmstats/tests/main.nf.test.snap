{
    "meles meles - fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "GCA_922984935.2.subset.fasta.gz.stats:md5,4f692268fb5b50290af7454a32744055"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e8d78f84b7fe3a16ef603a651edc2420"
                ],
                "stats": [
                    [
                        {
                            "id": "test"
                        },
                        "GCA_922984935.2.subset.fasta.gz.stats:md5,4f692268fb5b50290af7454a32744055"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e8d78f84b7fe3a16ef603a651edc2420"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-09-01T15:47:01.277673226"
    },
    "meles meles - fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.stats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,25c291e405becd2173f48a5130f5e03c"
                ],
                "stats": [
                    [
                        {
                            "id": "test"
                        },
                        "test.stats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,25c291e405becd2173f48a5130f5e03c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-09-01T15:47:06.899586868"
    }
}